### ISSUE TITLE
Metadata annotations are not propagated to created resources

### ISSUE THREAD

**Issue by kylegc** [appsody/appsody-operator#174]
### What did you do?

Deployed an appsody project with the operator (`appsody deploy`)

### What did you expect to see?

`kubectl describe pod my-project` properly shows the labels from the AppsodyApplication, but it does not show the annotations.

### What did you see instead?

I want to see all of the annotations from AppsodyApplication on all of the resources created by the operator.

### Environment

* Appsody CLI version: 0.4.8

* Appsody Stack: nodejs-express

* Kubernetes version information:
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.4", GitCommit:"f49fa022dbe63faafd0da106ef7e05a29721d3f1", GitTreeState:"clean", BuildDate:"2018-12-14T07:10:00Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.4", GitCommit:"67d2fcf276fcd9cf743ad4be9a9ef5828adc082f", GitTreeState:"clean", BuildDate:"2019-09-18T14:41:55Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by navidsh** [comment]
@kylegc We haven't implemented propagating annotations, we only support propagating labels. Is this something you want to be done urgently? Or is it okay to wait until the `0.3.0` released in a few weeks?

**Comment by leochr** [comment]
@kylegc FYI, in case you are looking to add annotations to `Service` resource only, we do have the `service.annotations` parameter.

**Comment by kylegc** [comment]
We can change this to a feature request, but I do believe it is urgent. 
The kAppNav dashboard will be looking for these annotations in the next Kabanaro release (next week?).

**Comment by leochr** [comment]
@kylegc We can implement it quickly and make it available by Monday (perhaps today). It'll have to be released as 0.2.2 version. Could you please test with 0.2.1 version and let us know if anything else is needed for the next Kabanero release. We can include all of those in 0.2.2. Thanks.

**Comment by leochr** [comment]
@JusteenR We already propagate labels down to the resources the operator creates. We need to do the same for annotations.
